### DISCUSSION TITLE
Seeking More Detailed Information for eMule Broadband Configuration Settings in preferences.ini

### DISCUSSION THREAD

**Discussion by Mika3578** [itlezy/eMule#4] [Q&A] [answered]
Hello,

I've been exploring the eMule Broadband edition and adjusting settings in the preferences.ini file. I've managed to understand a few of the parameters, but I would appreciate some more clarity on all of them, especially their minimum and maximum values.

The parameters I'm interested in are:

BBMaxUpClientsAllowed
BBMaxUploadTargetFillPerc
BBSlowRateTolerancePerc
BBSessionMaxTrans
BBSessionMaxTime
BBSlowDownloaderSampleDepth
BBUploadClientMaxDataRate
BBBoostLowRatioFiles
BBBoostLowRatioFilesBy
BBBoostFilesSmallerThan
BBDeboostLowIDs
BBDeboostHighRatioFiles
BBAutoFriendManagement
For each of these settings, could you please provide the following information:

A brief description of what each setting controls.
The minimum and maximum acceptable values for each setting.
The default value if not explicitly specified in the ini file.
Any impact of extreme values (either low or high) on the application's performance.
Your help in understanding these parameters would be greatly appreciated. This will not only help me but also others who are trying to fine-tune their eMule Broadband settings for optimal performance.

Thank you in advance for your time and assistance.

**Comment by edelkas** [comment]
These are all explained in the main [README](https://github.com/itlezy/eMule?tab=readme-ov-file#broadband-settings). The only one whose purpose is not clear, in my opinion, is `BBUploadClientMaxDataRate`. It's also not entirely clear what the "points" measure exactly when boosting or deboosting files.

**Comment by itlezy** [comment, marked as answer]
Good evening,

A lot has changed since the early broadband settings were introduced. We have been doing **lots** of live network analysis and adding instrumentation around the parts that matter most in real sessions:

- upload slot recycling
- elastic upload slot limit base + tolerance %
- download buffering and i/o optimization
- endgame behavior
- upload/download balance
- bad peer handling based on observed behavior

The direction is to tune the client from measured behavior, not from folklore settings.

For RC2, the focus is stabilization and making the broadband behavior easier to reason about. More detailed notes will follow on how slot recycling, buffering, and endgame strategy are being tuned.

We are also working on `emulebb-rust`, a new client written from scratch by the team. That is a separate long-term effort and not blocking the current RC2 work. Yes, this is real, not just a slop.

Follow the org page for updates. More technical notes are coming.
